crates/opt8n/src/opt8n.rs

```diff
@@ -97,6 +99,30 @@ impl Opt8n {
         Ok(())
     }
 
+    /// Run a Forge script with the given arguments, and generate an execution fixture
+    /// from the broadcasted transactions.
+    pub async fn run_script(&mut self, script_args: Box<ScriptArgs>) -> Result<()> {
+        let mut new_block = self.eth_api.backend.new_block_notifications();
+        self.eth_api.anvil_set_interval_mining(12)?;
+
+        let f = async {
+            let new_block = new_block.next();
+            join!(
+                new_block.into_future(),
+                script_args.run_script().map_err(Error::from)
+            )
+        };
+
+        if let (Some(new_block), _) = f.await {
+            tracing::info!("New block: {:?}", new_block);
+            if let Some(block) = self.eth_api.backend.get_block_by_hash(new_block.hash) {
+                self.generate_execution_fixture(block).await?;
+            }
+        }
+
+        Ok(())
```

Comment:
Or does the script require blocks to be actively mined in the background?

Reply:
It might be possible to run the script with the `--broadcast` flag and point it to the anvil endpoint generated by `opt8n`. If this works, after the script has run we could mine all of the transactions with `mine_block`. 

At first glance though, it looks like running forge script with `--broadcast` will wait for transaction receipts before `run_script()` completes.

Another approach would be to run the script normally and read the transactions from the `Cache` dir after the script completes. Then we could spin up a new anvil instance and replay the transactions.  Let me know if you have any other thoughts.

Reply:
> It might be possible to run the script with the --broadcast flag and point it to the anvil endpoint generated by opt8n. If this works, after the script finishes we could mine all of the transactions with mine_block.

Yeah this is the current functionality -- and how was testing it last night. 

Re synchrouous block mining: I honestly think this is a better approach -- reason I went with interval mining was mostly because the foundry thread was blocking waiting for transaction receipts & spamming the CLI with noise. Also wasn't positive what a difinitive signal would be to know when to call `mine_block` i.e. knowing at what point all tx's from the script have been broadcasted to Anvil.